runtime-modules/forum/src/lib.rs

```diff
@@ -790,6 +787,38 @@ decl_module! {
             Ok(())
         }
 
+        /// Update category
+        fn update_thread_archival_status(origin, actor: PrivilegedActor<T>, category_id: T::CategoryId, thread_id: T::ThreadId, new_archival_status: bool) -> dispatch::Result {
+            // Ensure data migration is done
+            Self::ensure_data_migration_done()?;
+
+            // Ensure actor can update category
+            let (category, thread) = Self::ensure_can_update_thread_archival_status(origin, &actor, &category_id, &thread_id)?;
+
+            if let Some(tmp_parent_category_id) = category.parent_category_id {
+                // Get path from parent to root of category tree.
+                let category_tree_path = Self::ensure_valid_category_and_build_category_tree_path(&tmp_parent_category_id)?;
+
+                if new_archival_status && Self::ensure_can_mutate_in_path_leaf(&category_tree_path).is_err() {
```

Comment:
I don't understand this check, it returns `Ok`, despite no storage state being updated, something seems wrong here, and also too complex. This would in all cases need a better explanation.

Reply:
This strange condition was what got left after I cut out other parts of old code around it. Now this condition gone because a situation it covered cannot occur anymore - we cannot archive thread in an already archived category, it will cause error because category cannot be mutated.